pkg/asset/store/store.go

```diff
@@ -83,6 +85,19 @@ func (s *storeImpl) Fetch(a asset.Asset, preserved ...asset.WritableAsset) error
 	if wa, ok := a.(asset.WritableAsset); ok {
 		return errors.Wrap(s.purge(append(preserved, wa)), "failed to purge asset")
 	}
+	// On the s390x/s390 platform, the igniton-config file will be converted to config
+	// drive image, so that the terraform can use it to inject metadata.
+	if runtime.GOARCH == "s390x" || runtime.GOARCH == "s390" && a.Config.Libvirt != nil {
+		if reflect.TypeOf(a).String() == "*bootstrap.Bootstrap" ||
```

Comment:
Why include this in the store and not the Ignition assets?

Reply:
@crawford Thanks, I agree. I will change in next update.

Reply:
It was brought up today in the CI/CD meeting that the node that does in the install is running on an x86_64 cluster, whereas the target node is s390x. I think you will need a check here to pull in the remote arch.

Reply:
I'm not sure if it's possible to determine the remote arch (cc @abhinavdahiya). We might have to define a new platform or maybe just add a field to the existing libvirt platform options (i.e. `UseConfigDrive` under the libvirt `Platform` struct).

Reply:
Remote arch can be determined by libvirt api which used by libvirt-go. However, installer should add the libvirt-go dependence, I don't think it's a good method.
Could we set an env variable (i.e. `TARGET`) to tell installer the target node's arch?

[terraform-provider-libvirt](https://github.com/dmacvicar/terraform-provider-libvirt) and [cluster-api-provider-libvirt](https://github.com/openshift/cluster-api-provider-libvirt) has libvirt-go, so they can expose apis to installer. 

In addition, we are trying if all these steps can be moved to terraform-provider-libvirt and cluster-api-provider-libvirt.